modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/operation/ComponentMessageProcessor.java

```diff
@@ -411,8 +412,34 @@ private void onEvent(CoreEvent event, ExecutorCallback executorCallback) {
       if (location != null) {
         ((DefaultFlowCallStack) event.getFlowCallStack())
             .pushCurrentProcessorPath(resolvedProcessorRepresentation);
-        sdkInternalContext.getPolicyToApply(location, eventId)
-            .process(event, operationExecutionFunction, () -> resolutionResult, location, executorCallback);
+
+        ExecutorCallback effectiveCallback = new ExecutorCallback() {
+
+          @Override
+          public void complete(Object o) {
+            CoreEvent policyResult = (CoreEvent) o;
+            if (isTargetWithPolicies(event)) {
```

Comment:
shouldn't this be the only place where `target` is handled? Why do need to know if a policy is applied?

Reply:
When the operation doesn't have a target/targetValue, the behaviour is the same we had before the change: the returnDelegate is used in [ComponentMessageProcessor#mapped](https://github.com/mulesoft/mule/blob/MULE-18375/modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/operation/ComponentMessageProcessor.java#L478), otherwise, [ComponentMessageProcessor#mapped](https://github.com/mulesoft/mule/blob/MULE-18375/modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/operation/ComponentMessageProcessor.java#L478) uses a ValueReturnDelegate and here, after `PolicyEventMapper#onOperationPolicyFinish` was executed, is used the appropriate returnDelegate.